### ISSUE TITLE
Android crashes: signal 31 (SIGSYS), code 1 (SYS_SECCOMP)

### ISSUE THREAD

**Issue by DentonGentry** [tailscale/tailscale#2478]
Starting July 3rd, the date of the 10.1 release, a new Android crash appeared:
```
*** *** *** *** *** *** *** *** *** *** *** *** *** *** *** ***
pid: 0, tid: 0 >>> com.tailscale.ipn <<<

backtrace:
  #00  pc 00000000002fc0d8  /data/app/~~XXX==/com.tailscale.ipn-XXX-XXX==/split_config.arm64_v8a.apk!lib/arm64-v8a/libgio.so (offset 0x1000)
```

<img width="1034" alt="Screen Shot 2021-07-20 at 8 46 22 PM" src="https://user-images.githubusercontent.com/1140756/126427595-33a9a980-91dc-4ab7-8a6f-534b0b5f5678.png">

It occurs at libgio.so (offset 0x1000). Offset 0x1000 seems suspiciously round.

July 3rd was the date of the 10.1 release, rev 47. The crash only happened 7 times on older releases, but almost 2000 times since.

There are several distinctive things about rev 47:
- it is the first production release I built. Prior releases were built by Elias.
- it is the first production release after a Gio update in 1.10.0. If the crash is in the updated libgio but occurs when exiting to update the app, rev 47 would be the first to see it. I bring this up because the crashes peaked when the 1.10.2 upgrades peaked and have declined since the 1.10.2 rollout finished.

**Comment by xindirama** [comment]
I wonder if this is what is biting me. Just upgraded Tailscale on my OEM Pixel from 1.10.1 to 1.11 using FDroid, and it just kept crashing, I could not make it work until I cleared all storage and then I was able to re-authenticate. But I kept being thrown back to the login screen, even though the device was showing up in the Tailscale control panel.

I downgraded to 1.10.2, again had to clear storage and re-authenticate, but now it seems to be working.

Slightly different behavior yesterday on another Android running LineageOS without Google Services. I could not make either 1.10.2 or 1.11 work, I had to downgrade all the way back to 1.10.1 before I could get it working again.

**Comment by DentonGentry** [comment]
Quite possibly, yes. Are you comfortable using adb?

We found one way to reproduce a SYS_SECCOMP crash in https://github.com/tailscale/tailscale/issues/2450, but the haven't been able to produce the repeated crashing you describe and which the Play Store metrics are reporting. We know it is SYS_SECCOMP, and we suspect the accept() system call is the problem, but we don't know that for sure.

If you're comfortable using ADB, you'd connect the USB cable and run: `adb logcat -s com.tailscale.ipn:*`
Then start the Tailscale app, let it crash once or twice, and send the logs that appear from adb.

If you don't want to paste it into a public bug report you can send it to support@tailscale.com

**Comment by DentonGentry** [comment]
I uploaded an Open Testing build about an hour ago which might resolve this.
1.11.151-d145c594ad6-g7ebedfd62a9

It fixes https://github.com/tailscale/tailscale/issues/2450, which might be the same underlying crash as we see here but in a different location.

You'd need to sign up for the Open Testing track at https://play.google.com/apps/testing/com.tailscale.ipn
Wait a few minutes, then go to the Play Store and check that it is offering to install 1.11.151-d145c594ad6-g7ebedfd62a9

**Comment by xindirama** [comment]
Where it exists on my devices, Google Play is everywhere disabled, so I regret I will not be able to test until this hits FDroid.

> If you don't want to paste it into a public bug report you can send it to support@tailscale.com
 
This e-mail address is being delivered to Google servers, so unless you have a GPG key for this e-mail address, it is not much of an alternative. I will have a look at the logcat output and see if I see anything.

**Comment by xindirama** [comment]
[samsung-log-1.11.txt](https://github.com/tailscale/tailscale/files/6872385/samsung-log-1.11.txt)
This log is from my first try with my old tablet, soon to go to pasture with the kids. LineageOS, zero google apps, Android 9. This time zero crashes, just repeated failures to login from the device side, despite an entry for the device appearing very reliably in the control panel for every login.

**Comment by DentonGentry** [comment]
I'm going to need to do something differently, it looks like 1.11.151 isn't building on F-Droid.

I need to get F-Droid to use either https://github.com/tailscale/go or go 1.17-rc1 to build, because we need golang.org/cl/313769 in the toolchain to fix this SYS_SECCOMP crash.

**Comment by DentonGentry** [comment]
Android release 1.12.1 is rolling out now in the Play Store. It updates the Go toolchain to https://github.com/tailscale/go, which automatically uses accept4() instead of accept() for Android. We'll monitor for occurrences of this issue with 1.12.1.

The F-Droid build is still not working. We've split it out into https://github.com/tailscale/tailscale/issues/2536 to work on.

**Comment by DentonGentry** [comment]
1.12.1 has had 0 of these crashes so far, even as it rolls out to more and more of the users/

**Comment by DentonGentry** [comment]
No SYS_SECCOMP crashes with 1.12.1. Calling this fixed.

**Comment by xindirama** [comment]
Yup, new version on FDroid also works on one of my devices.

**Comment by DentonGentry** [comment]
Excellent. The F-Droid builder is a bit backlogged, but the 1.11.151 build it just released is awfully close to what went into 1.12.1.
